fetch.js

```diff
@@ -1,23 +1,47 @@
-(function(self) {
+(function (root, factory) {
   'use strict';
 
-  if (self.fetch) {
-    return
+  if (typeof define === 'function' && define.amd) {
+    // AMD. Register as an anonymous module.
+    define([], factory)
+  } else if (typeof module === 'object' && module.exports) {
+    // Node. Does not work with strict CommonJS, but
+    // only CommonJS-like environments that support module.exports,
+    // like Node.
+    module.exports = factory()
+  } else {
+    // Browser globals (root is window)
+    root.fetchPolyfill = factory()
+    if (!root.fetch) {
+      for (var attr in root.fetchPolyfill) {
+        root[attr] = root.fetchPolyfill[attr]
+      }
+    }
+  }
+}(this, function () {
+  'use strict';
+
+  var main = {};
+
+  try {
+    main = window
+  } catch(e) {
+    main = global
   }
 
   var support = {
-    searchParams: 'URLSearchParams' in self,
```

Comment:
I cannot name it self because I already defined it later to avoid diffs on more important stuff (since what was self is now split between `self` = what is exported and `main` = global object). I'll rename it `root` while doing the other changes you suggest so it keeps the same nomenclature as the UMD definition.